examples/train_cifar10.py

```diff
@@ -85,28 +89,52 @@ def dataloader():
     config.merge_from_file('configs/pytorch_trainer/trainer.yaml')
     config.merge_from_list(['HOOK.CheckpointHook.interval', 3])
 
+    # create working directory
+    work_dir = './dev/trainer/'
+    work_dir = create_work_dir(work_dir)
+
+    # logger
+    logger = get_logger(work_dir)
+
+    # set random seed
+    seed = config.trainer.seed
+    deterministic = config.trainer.deterministic
+    set_random_seed(logger,
+                    seed=seed,
+                    deterministic=deterministic)
+
+    # get device
+    gpu_ids = config.trainer.gpu_ids
+    device = get_device(logger,
+                        gpu_ids=gpu_ids,
+                        deterministic=deterministic)
+
```

Comment:
Why we place `create_work_dir`, `get_logger`, `set_random_seed`, `get_device` out of `EpochBasedTrainer`

Reply:
Dependency Injection